rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/guid_utils.hpp

```diff
@@ -55,6 +56,35 @@ copy_from_fastrtps_guid_to_byte_array(
   memcpy(&guid_byte_array[prefix_size], &guid.entityId, guid.entityId.size);
 }
 
+struct hash_fastrtps_guid
+{
+  std::size_t operator()(const eprosima::fastrtps::rtps::GUID_t & guid) const
+  {
+    union u_convert {
+      uint8_t plain_value[sizeof(guid)];
+      uint32_t plain_ints[4];
```

Comment:
@MiguelCompany 
```suggestion
      uint32_t plain_ints[sizeof(guid)/sizeof(uint32_t)];
```
?

Reply:
Yeah. But we should also check we have 4 integers, as the calculation below is assuming that.

Should I also add a check to the `staic_assert` below for `sizeof(guid) == 16`?

Reply:
> But we should also check we have 4 integers, as the calculation below is assuming that.

That is a good point. 

> Should I also add a check to the staic_assert below for sizeof(guid) == 16?

Fair enough.